apps/opt.c

```diff
@@ -326,6 +326,33 @@ int opt_int(const char *value, int *result)
     return 1;
 }
 
+static void opt_number_error(const char *v)
+{
+    int i = 0;
+    struct strstr_pair_st {
+        char *prefix;
+        char *name;
+    } b[] = {
+        {"0x", "a hexadecimal"},
+        {"0X", "a hexadecimal"},
+        {"0", "an octal"},
+        {NULL}
+    };
+
+    while (b[i].prefix != NULL) {
+        if (strncmp(v, b[i].prefix, strlen(b[i].prefix)) == 0) {
+            BIO_printf(bio_err,
+                       "%s: Can't parse \"%s\" as %s number\n",
+                       prog, v, b[i].name);
+            break;
+        }
+        i++;
+    }
+    if (i == sizeof(b) / sizeof(struct strstr_pair_st) - 1)
```

Comment:
As a side note, I'd say `if (b[i].prefix == NULL)` would be more appropriate here. Because it matches `while` expression.